bin/git-delete-branch

```diff
@@ -1,16 +1,94 @@
 #!/usr/bin/env bash
 
-# Assert there is at least one branch provided
-test -z $1 && echo "branch required." 1>&2 && exit 1
-
-for branch in "$@"
-do
-  remote=$(git config branch.$branch.remote)
-  test -z $remote && remote="origin"
-  ref=$(git config branch.$branch.merge)
-  test -z $ref && ref="refs/heads/$branch"
-
-  git branch -D $branch
-  git branch -d -r $remote/$branch
-  git push $remote :$ref
+set -u
+
+usage()
+{
+    local command="git delete-branch"
+    cat << EOF
+Usage:
+    ${command} [-r <remote[:remote branch name]>] <branch>
+    ${command} -h
+
+Delete local and remote branches. If the remote switch (-r) is provided, then
+use the given remote and remote branch. Otherwise, the remote branch is found
+by first trying the upstream tracking branch. If there is no tracking branch
+configured, then deletion will not succeed unless provided explicitly. If the
+remote branch and local branch do not share a name, the remote branch name can
+be included with the remote switch.
+EOF
+}
+
+remote=
+remote_branch=
+
+while getopts "r:h" opt; do
+    case "$opt" in
+        r)
+            remote="$(echo ${OPTARG} | cut -d ':' -f 1)"
+            remote_branch="$(echo ${OPTARG} | cut -s -d ':' -f 2)"
```

Comment:
> Better to use parameter expansion instead.

Hmm I thought using cut would be more clear. So is this what you're suggestion: 

So instead of: 
```sh
remote="$(echo ${OPTARG} | cut -d ':' -f 1)"
remote_branch="$(echo ${OPTARG} | cut -s -d ':' -f 2)"
```

you want: 

```sh
remote=${OPTARG%%:*}
remote_branch=${OPTARG##*:}
```